app/factory/Layer.js

```diff
@@ -190,12 +190,35 @@ Ext.define('CpsiMapview.factory.Layer', {
             visible: layerConf.openLayers.visibility,
             minResolution: layerConf.openLayers.minResolution,
             maxResolution: layerConf.openLayers.maxResolution,
-            opacity: layerConf.openLayers.opacity
+            opacity: layerConf.openLayers.opacity,
+            toolTipConfig: layerConf.tooltipsConfig
         });
 
         if (layerConf.tooltipsConfig) {
-            //TODO has to be implemented with
-            //     https://github.com/meggsimum/cpsi-mapview/issues/27
+            // create a custom toolitp for this layer
+            var toolTip = Ext.create('CpsiMapview.view.layer.ToolTip', {
+                toolTipConfig: layerConf.tooltipsConfig,
+                layer: wfsLayer
+            });
+            wfsLayer.toolTip = toolTip;
+
+            // show / hide on appropriate events
+            mapPanel.on('cmv-map-pointerrest', function(hoveredObjs, evt) {
```

Comment:
Do we need to check if the hovered feat is from this layer?

Reply:
This is implicitly done in Line 214. The tooltip, which is rendered, is bound to the layer and is only available if the layer is 'hoverable'.

Reply:
What I meant is the following: if we duplicate the layer GAS WFS and call it GAS WFS 2, and put in a log in line 2017 which prints out the `hoveredObjs`, you'll see, that we get two logs (expected), but both of them have two features (rather unexpected, right?). A possible (untested) fix could be to check `hoveredObj.layer` against `wfsLayer`. Is this more clear now?

Reply:
Done in 51cfef734b52.